controllers/clusteroperator.go

```diff
@@ -0,0 +1,222 @@
+	return osconfigv1.ClusterOperatorStatusCondition{
+		Type:               conditionType,
+		Status:             conditionStatus,
+		LastTransitionTime: metav1.Now(),
+		Reason:             reason,
+		Message:            message,
+	}
+}
+
+// getOperandVersions returns current version
+func getOperandVersions() []osconfigv1.OperandVersion {
+	operandVersions := []osconfigv1.OperandVersion{}
+	if releaseVersion := os.Getenv("RELEASE_VERSION"); len(releaseVersion) > 0 {
+		operandVersions = append(operandVersions, osconfigv1.OperandVersion{Name: "operator", Version: releaseVersion})
+	}
+	return operandVersions
+}
+
+//syncStatus applies the new condition to the mao ClusterOperator object.
+func (r *ProvisioningReconciler) syncStatus(co *osconfigv1.ClusterOperator, conds []osconfigv1.ClusterOperatorStatusCondition) error {
+	for _, c := range conds {
+		v1helpers.SetStatusCondition(&co.Status.Conditions, c)
+	}
+
+	_, err := r.OSClient.ConfigV1().ClusterOperators().UpdateStatus(context.Background(), co, metav1.UpdateOptions{})
+	return err
+}
+
+// updateCOStatusProgressing updates the ClusterOperator's status to Progressing
+func (r *ProvisioningReconciler) updateCOStatusProgressing() error {
+	var isProgressing osconfigv1.ConditionStatus
+	var message string
+
+	co, err := r.getOrCreateClusterOperator()
+	if err != nil {
+		glog.Errorf("Failed to get or create ClusterOperator: %v", err)
+		return err
+	}
+
+	currentVersions := co.Status.Versions
+	targetVersions := getOperandVersions()
+
+	if !reflect.DeepEqual(targetVersions, currentVersions) {
+		glog.V(2).Info("Syncing status: progressing")
+		r.EventRecorder.Eventf(co, v1.EventTypeNormal, "Status upgrade", message)
+		isProgressing = osconfigv1.ConditionTrue
+		message = fmt.Sprintf("Progressing towards target version")
+	} else {
+		glog.V(2).Info("Syncing status: re-syncing")
+		isProgressing = osconfigv1.ConditionFalse
+		message = fmt.Sprintf("Re-sync running")
+	}
+
+	conds := []osconfigv1.ClusterOperatorStatusCondition{
+		setStatusCondition(osconfigv1.OperatorAvailable, osconfigv1.ConditionFalse, "Starting up", "Before start of Metal3 deployment"),
+		setStatusCondition(osconfigv1.OperatorProgressing, isProgressing, string(ReasonSyncing), message),
+	}
+
+	return r.syncStatus(co, conds)
+}
+
+// updateCOStatusAvailable updates the ClusterOperator's status to Available
+func (r *ProvisioningReconciler) updateCOStatusAvailable() error {
+	message := "Metal3 pod available"
+
+	co, err := r.getOrCreateClusterOperator()
+	if err != nil {
+		glog.Errorf("Failed to get or create ClusterOperator: %v", err)
+		return err
+	}
+
+	conds := []osconfigv1.ClusterOperatorStatusCondition{
+		setStatusCondition(osconfigv1.OperatorAvailable, osconfigv1.ConditionTrue, string(ReasonComplete), message),
+	}
+
+	return r.syncStatus(co, conds)
+}
+
+// updateCOStatusDegraded updates the ClusterOperator's status to Degraded
+func (r *ProvisioningReconciler) updateCOStatusDegraded() error {
```

Comment:
Can we add this method here when we are ready to call it from Reconcile()? We can also add unit tests at that time.

Reply:
These snuck back into the latest commit after I split the commits a short while ago. I'll remove this since we are not using it in reconcile at this time.